pkg/util/node/node.go

```diff
@@ -30,6 +30,13 @@ import (
 	"k8s.io/klog/v2"
 )
 
+// Labels definitions.
+const (
+	// NodePoolPodRangeLabelPrefix is the prefix for the default Pod range
+	// name for the node and it can be different with cluster Pod range
+	NodePoolPodRangeLabelPrefix = "cloud.google.com/gke-np-default-pod-range"
```

Comment:
Do we need to ensure GKE nodes have this label present before we merge this PR?

Reply:
We can check this in but not cut a new CCM version. That could be risky though. We should do this right and wait for the version trait code to roll out. We could also make a check and avoid the logic altogether.

Reply:
Agree, I would prefer add a check to split the logic. Maybe introducing a boolean constant like `enableMPC` to 'turn off' the feature for now and 'turn on' after the version trait out and right before cutting a new CCM version. We can put it after the node event handler and in the GNP controller reconcile loop. Without the actual label, node informer will never add the default GNP to the work queue, while for GNP informer the constant is false then we don't call the `c.syncPodRanges`.

The following PR about the nodeipam controller change to enable node pool using multi pod cidr can be guarded under the same constant as well.

Reply:
we synced offline, and basically we have all the components to re-create default GNP from scratch. Having that we can then support seamlessly all the Node versions, if it will have label we will add it the cluster-default, if nothing there, we will always have the cluster-default. This means as well that `params.Spec.PodIPv4Ranges = nil` will never be a case, and at least will hold cluster-default range.

Reply:
Done, please review again, thanks!